app/routes/_layout+/assets.$assetId.duplicate.tsx

```diff
@@ -0,0 +1,150 @@
+import type { ActionArgs, LoaderArgs } from "@remix-run/node";
+import { json, redirect } from "@remix-run/node";
+import { Form, useLoaderData, useNavigation } from "@remix-run/react";
+import { AssetImage } from "~/components/assets/asset-image";
+import Input from "~/components/forms/input";
+import { Badge, Button } from "~/components/shared";
+import { Spinner } from "~/components/shared/spinner";
+import { db } from "~/database";
+import { duplicateAsset } from "~/modules/asset";
+import { requireAuthSession } from "~/modules/auth";
+import styles from "~/styles/layout/custom-modal.css";
+import {
+  assertIsPost,
+  isFormProcessing,
+  userFriendlyAssetStatus,
+} from "~/utils";
+import { MAX_DUPLICATES_ALLOWED } from "~/utils/constants";
+import { sendNotification } from "~/utils/emitter/send-notification.server";
+import { ShelfStackError } from "~/utils/error";
+
+export const loader = async ({ params }: LoaderArgs) => {
+  const assetId = params.assetId as string;
+  const asset = await db.asset.findUnique({ where: { id: assetId } });
+  if (!asset) {
+    throw new ShelfStackError({ message: "Asset Not Found", status: 404 });
+  }
+
+  return json({
+    showModal: true,
+    asset,
+  });
+};
+
+export const action = async ({ request, params }: ActionArgs) => {
+  assertIsPost(request);
+
+  const { userId } = await requireAuthSession(request);
+  const assetId = params.assetId as string;
+  const asset = await db.asset.findUnique({
+    where: { id: assetId },
+    include: { custody: true },
+  });
+  if (!asset) {
+    throw new ShelfStackError({ message: "Asset Not Found", status: 404 });
+  }
+
+  const formData = await request.formData();
+  const amountOfDuplicates =
+    (formData.get("amountOfDuplicates") as string) ?? 1;
+
+  const allowedDuplicates = Math.min(
```

Comment:
This works technically but if I understand it correctly it will create a very bad UX as it will just set the amount to the maxAllowed and go forward without informing the user.
What you need to do here is the following:

1. Create schema for your form using zod, like we do in all our other forms
2. Ensure that the value of the field there is not bigger than the max and create proper errors for it.